src/uwtools/rocoto.py

```diff
@@ -179,27 +201,28 @@ def _add_task_dependency_operand_operator(self, e: Element, config: dict) -> Non
         :param e: The parent element to add the new element to.
         :param config: Configuration data for this element.
         """
-
         operands, operators, strequality = self._dependency_constants
-
-        for tag, block in config.items():
+        for tag, subconfig in config.items():
             tag, _ = self._tag_name(tag)
             if tag in operands:
-                self._set_attrs(SubElement(e, tag), block)
+                if tag == STR.taskdep:
+                    self._set_attrs(SubElement(e, tag), subconfig)
+                else:
+                    self._add_compound_time_string(e, config, tag)
```

Comment:
This was something I also struggled with when debating how much to breakdown methods/functions into smaller more concentrated ideas. I think the trade off we were seeing before was this would violate DRY. So the question remains, when does this begin to do too much that it out ways the benefit of not repeating code.